packages/palette/src/elements/Image/LazyImage.tsx

```diff
@@ -64,14 +65,63 @@ export const LazyImage: React.FC<LazyImageProps> = ({
     borderRadius,
     style,
     onError,
+    blurhash,
     ...containerProps
   } = props
 
   if (preload) {
-    return <ImageComponent {...props} />
+    return (
+      <SkeletonImage
+        width={width}
+        height={height}
+        borderRadius={borderRadius}
+        blurhash={hideBlurhash ? undefined : blurhash}
+        {...containerProps}
+      >
+        <ImageComponent onLoad={() => setHideBlurhash(true)} {...props} />
+      </SkeletonImage>
+    )
+  }
+
+  const handleLoad = () => {
+    setImageLoaded(true)
+    setTimeout(() => {
+      setHideBlurhash(true)
+    }, 500)
```

Comment:
I updated the handler to remove the BlurHash canvas when the image is loaded and to remove the flashing

Here's a comparison table:

| WIth new State | Without new State | Keeping the canvas in the background|
| -- | -- | -- |
| <video src="https://github.com/artsy/palette/assets/20655703/0dfeccce-3145-41a8-9561-74b2ea4cacec" /> | <video src="https://github.com/artsy/palette/assets/20655703/ddde351d-f7e0-4b83-880d-0826eaf30408" /> | <video src="https://github.com/artsy/palette/assets/20655703/241a39af-92f6-444a-97fb-7377458b1505" /> |

Reply:
Well the problem is timing the transition. This component is 🗑️ — ideally we'd use the web animations API to have a real callback.